tests/device/test_http_client/test_http_client.ino

```diff
@@ -48,17 +50,20 @@ TEST_CASE("HTTP GET & POST requests", "[HTTPClient]")
     }
     {
         // can do two POST requests with one HTTPClient object (#1902)
+        WiFiClient client;
         HTTPClient http;
-        http.begin(getenv("SERVER_IP"), 8088, "/");
+        http.begin(client, getenv("SERVER_IP"), 8088, "/");
         http.addHeader("Content-Type", "text/plain");
         auto httpCode = http.POST("foo");
         Serial.println(httpCode);
         REQUIRE(httpCode == HTTP_CODE_OK);
+/* TBD: This test is broken because end() nulls the client*
```

Comment:
Thanks for doing the changes, you beat me to it.  I noticed this failing while making my own device tests for two PRs.

Any reason you don't want to reset the client after `.end()` and run the POST?  I'm not sure the test is caring about the `.end()` behavior here so much as a "Can I do a successful POST"?

Reply:
The comment in the check says two back to back POSTs and points to this ticket: https://github.com/esp8266/Arduino/issues/1902 which does POST();end();POST(); and that's broken.  Should we test both old and new APIs?  And both BearSSL and AXTLS? (or will they go away soon enough?)  In any case the specific usage listed in the above ticket is broken.

Reply:
I'm about to commit an update that will check that the second POST() after end() without another begin() does not crash. This includes a fix in HTTPClient::connect() where the crash was ocuring.